src/components/AvatarCropModal/libs/imageManipulator/index.js

```diff
@@ -0,0 +1,122 @@
+import _ from 'underscore';
+
+function getContext(canvas) {
+    return canvas.getContext('2d');
+}
+
+function sizeFromAngle(width, height, angle) {
+    const radians = (angle * Math.PI) / 180;
+    let c = Math.cos(radians);
+    let s = Math.sin(radians);
+    if (s < 0) {
+        s = -s;
+    }
+    if (c < 0) {
+        c = -c;
+    }
+    return {width: (height * s) + (width * c), height: (height * c) + (width * s)};
+}
+
+function rotate(canvas, degrees) {
+    const {width, height} = sizeFromAngle(canvas.width, canvas.height, degrees);
+
+    const result = document.createElement('canvas');
+    result.width = width;
+    result.height = height;
+
+    const context = getContext(result);
+
+    // Set the origin to the center of the image
+    context.translate(result.width / 2, result.height / 2);
+
+    // Rotate the canvas around the origin
+    const radians = (degrees * Math.PI) / 180;
+    context.rotate(radians);
+
+    // Draw the image
+    context.drawImage(canvas, -canvas.width / 2, -canvas.height / 2, canvas.width, canvas.height);
+
+    return result;
+}
+
+function crop(canvas, options) {
+    // ensure values are defined.
+    let {
+        originX = 0, originY = 0, width = 0, height = 0,
+    } = options;
+    const clamp = (value, max) => Math.max(0, Math.min(max, value));
+
+    // lock within bounds.
+    width = clamp(width, canvas.width);
+    height = clamp(height, canvas.height);
+    originX = clamp(originX, canvas.width);
+    originY = clamp(originY, canvas.height);
+
+    // lock sum of crop.
+    width = Math.min(originX + width, canvas.width) - originX;
+    height = Math.min(originY + height, canvas.height) - originY;
+
+    const result = document.createElement('canvas');
+    result.width = width;
+    result.height = height;
+
+    const context = getContext(result);
+    context.drawImage(canvas, originX, originY, width, height, 0, 0, width, height);
+
+    return result;
+}
+
+function getResults(canvas) {
```

Comment:
This function name is too generic, can you please rename it to have a more specific meaning?

Reply:
Renamed to convertCanvasToFile